Sources/MongoSwift/Operations/AggregateOperation.swift

```diff
@@ -85,33 +94,56 @@ internal struct AggregateOperation<CollectionType: Codable, OutputType: Codable>
         let opts = try encodeOptions(options: self.options, session: session)
         let pipeline: BSONDocument = ["pipeline": .array(self.pipeline.map { .document($0) })]
 
-        let result: OpaquePointer = self.collection.withMongocCollection(from: connection) { collPtr in
-            pipeline.withBSONPointer { pipelinePtr in
-                withOptionalBSONPointer(to: opts) { optsPtr in
-                    ReadPreference.withOptionalMongocReadPreference(from: self.options?.readPreference) { rpPtr in
-                        guard let result = mongoc_collection_aggregate(
-                            collPtr,
-                            MONGOC_QUERY_NONE,
-                            pipelinePtr,
-                            optsPtr,
-                            rpPtr
-                        ) else {
-                            fatalError(failedToRetrieveCursorMessage)
+        return try pipeline.withBSONPointer { pipelinePtr in
+            try withOptionalBSONPointer(to: opts) { optsPtr in
+                try ReadPreference.withOptionalMongocReadPreference(from: self.options?.readPreference) { rpPtr ->
+                    MongoCursor<OutputType> in
```

Comment:
is this `rpPtr -> MongoCursor<OutputType>` annotation necessary? The formatting is kinda funky since it lines up with the rest of the statements in the closure, so it'd be nice if we could just delete it. If it is necessary, then can we add a line break before the call to `withOptionalMongocReadPreference` to try to get the entire closure type annotation on a single line?